core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -345,6 +346,17 @@ public static String sha512(ByteString string)  {
     return DigestUtils.sha512Hex(string.getBytes());
   }
 
+  /** SQL {@code REGEXP_CONTAINS(value, regexp)} function.
+   * Throws a runtime exception for invalid regular expressions.*/
+  public static boolean regexpContains(String value, String regex) {
+    try {
+      Pattern regexp = Pattern.compile(regex);
```

Comment:
If you're going to stick with Java library, I think it would help to add a comment here or elsewhere linking to the case where that decision was made for future users/devs

Reply:
Thanks for the pointer Tanner, added a note with issue link to the method description.

Reply:
I wouldn't link to jira from the source code. Comments in the source code should be self-contained.

(I know we add jira links in bugs. But that's because we *removed* a bug, and we don't want to leave a trail of mistakes in the source code.)

This is a decision affects the specification of the function, not just its implementation. So I think the right place for the comment is in `big-query.iq`. You could have another comment in the body of `regexpContains` if you think maintainers might be confused why we're using Java regex.

Reply:
I think it makes sense to have one in both place because If someone were to write a query using regex that was valid in the re2 library and not Java, if it used Calcite's implementation then the error would get thrown from `regexpContains` and the user/developer would be able to reference that comment for clarification on the library decision.

Reply:
Sure I think it makes sense to add a comment in the method body too, just wanted to point out anything that works in RE2 should be processed identically by java regex as well, which should cover 99.9% of queries. Only some invalid regex errors caught by RE2 aren't caught here so Calcite might produce an incorrect output for a few corner cases, which are technically incorrect regex inputs so they shouldn't be used in a query anyway.